kindcat_custom.cue

```diff
@@ -10,6 +98,79 @@ package kindsys
 Custom: S={
 	_sharedKind
 
-	lineage: { name: S.machineName }
+	// pluginID is the unique identifier of the plugin which owns this Custom kind
+	pluginID: =~"^([a-z][a-z0-9-]*[a-z0-9])$"
```

Comment:
Thoughts on whether this should be a top-level field, or live _only_ in the `plugin` trait (or even if the `plugin` trait should disappear, and only the `pluginID` reference should remain)?

Reply:
i think we need to firm up the notion of a trait to really decide.

For the CUE part of it, i've been imagining traits as discrete sets of fields that can be composed into kind category definitions. Those sets - which in this PR i would imagine include both the [top-level props](https://github.com/grafana/kindsys/pull/9/files#diff-8cc62ac75bb235b29e2f781042da0b46a914dcb9dc9ebc2a86ea3abac2d89977R122-R160) and [the joinschema](https://github.com/grafana/kindsys/pull/9/files#diff-8cc62ac75bb235b29e2f781042da0b46a914dcb9dc9ebc2a86ea3abac2d89977R38-R89) - would then be imported into the root package and composed into the kind category definitions in as simple a manner possible. i'm thinking of that as valuable because as a reader/maintainer of the kind cat definitions, i want to be able to easily see and reason about how the trait is composed across kind categories, basically independent of what's in the trait itself.

Given the above, and because we actually need plugins to be [a container/wrapper _around_ kind definitions](https://github.com/grafana/grafana/blob/e70d623f90687f2b6c685dad21fc6d4d9b7d0077/pkg/plugins/pfs/grafanaplugin.cue), they fall at least partly outside my current operating concept of "trait". Happy to change my mind here if we find something that feels right, though!

Reply:
Makes sense to me, I think we can ditch the `plugin` trait in favor of just having the `pluginID` field as the top-level prop for Custom.

Reply:
Based on discussions/input yesterday and the need for grouping all kinds (core, custom, composable) to fit in a schema registry I think `group` or similar makes more sense as well. Currently working on a doc that hopefully should propose how all of this are supposed to fit together in regards to a schema registry and plugins.

Reply:
I've updated it to be `group` in the top-level, just so we don't need to change as much later, and it doesn't explicitly reference plugins within the kind itself.